pkg/volume/iscsi/iscsi.go

```diff
@@ -213,12 +213,29 @@ func (plugin *iscsiPlugin) newUnmounterInternal(volName string, podUID types.UID
 }
 
 func (plugin *iscsiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
+	mounter := plugin.host.GetMounter(plugin.GetPluginName())
+	pluginDir := plugin.host.GetPluginDir(plugin.GetPluginName())
+	sourceName, err := mounter.GetDeviceNameFromMount(mountPath, pluginDir)
+	if err != nil {
+		return nil, err
+	}
+	str := strings.Split(sourceName, "-lun-")
+	index := strings.Index(str[0], "-")
```

Comment:
validate str array first before use

Reply:
@rootfs I put it in line below(L224), if str is empty or not len(2), will return error, str[0] (L223) will not cause panic here.